GitUI/CommandsDialogs/UserScriptContextMenuExtensions.cs

```diff
@@ -10,90 +10,104 @@ namespace GitUI.CommandsDialogs
+                    hostMenuItem.DropDown.Items.Add(item);
                 }
             }
 
-            void AddOwnScripts()
+            if (lastIndex != contextMenu.Items.Count)
             {
-                var lastIndex = contextMenu.Items.Count;
+                contextMenu.Items.Insert(lastIndex, new ToolStripSeparator());
+            }
 
-                var scripts = ScriptManager.GetScripts();
-                foreach (var script in scripts)
-                {
-                    if (!script.Enabled)
-                    {
-                        continue;
-                    }
-
-                    var item = new ToolStripMenuItem
-                    {
-                        Text = script.Name,
-                        Name = script.Name + "_ownScript",
-                        Image = script.GetIcon(),
-                        ShortcutKeyDisplayString = Hotkeys.Value?.FirstOrDefault(h => h.Name == script.Name)?.KeyData.ToShortcutKeyDisplayString()
-                    };
-
-                    item.Click += (s, e) =>
-                    {
-                        string scriptKey = script.Name;
-                        scriptInvoker(scriptKey);
-                    };
-
-                    if (script.AddToRevisionGridContextMenu)
-                    {
-                        contextMenu.Items.Add(item);
-                    }
-                    else
-                    {
-                        hostMenuItem.DropDown.Items.Add(item);
-                    }
-                }
+            bool showScriptsMenu = hostMenuItem.DropDown.Items.Count > 0;
+            hostMenuItem.Visible = showScriptsMenu;
+        }
+
+        /// <summary>
+        ///  Removes user scripts from the <paramref name="contextMenu"/>, or under <paramref name="hostMenuItem"/>,
+        ///  if scripts are marked as <see cref="ScriptInfo.AddToRevisionGridContextMenu"/>.
+        /// </summary>
+        /// <param name="contextMenu">The context menu to add user scripts too.</param>
+        /// <param name="hostMenuItem">The menu item to which to add user scripts marked as <see cref="ScriptInfo.AddToRevisionGridContextMenu"/>.</param>
+        public static void RemoveUserScripts(this ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem)
+        {
+            contextMenu = contextMenu ?? throw new ArgumentNullException(nameof(contextMenu));
+            hostMenuItem = hostMenuItem ?? throw new ArgumentNullException(nameof(hostMenuItem));
+
+            RemoveOwnScripts(contextMenu, hostMenuItem);
+        }
 
-                if (lastIndex != contextMenu.Items.Count)
+        private static void RemoveOwnScripts(ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem)
+        {
+            hostMenuItem.DropDown.Items.Clear();
+
+            var list = contextMenu.Items.Cast<ToolStripItem>().ToList();
+
+            foreach (var item in list)
+            {
+                if (item.Name.Contains(ScriptNameSuffix))
```

Comment:
Use LINQ and more precise condition `.Where(item => item.Name.EndsWith(ScriptNameSuffix)`.
Can `.ToList()` be omitted?

Reply:
> Can .ToList() be omitted?

Can't iterate and remove item from collection.
We need to create a copy of the collection.